sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -25,7 +25,22 @@
 class BaseLoss(ABC):
     """Base class for a loss."""
 
-    def init_gradients_and_hessians(self, n_samples, prediction_dim):
+    @abstractmethod
+    def __call__(self, y_true, raw_predictions):
+        """Return the loss for each value
+        """
+
+    def get_average_loss(self, y_true, raw_predictions, sample_weight):
+        """Return the average loss, weighted if sample_weight is not None
+        """
+        if sample_weight is None:
+            return self(y_true, raw_predictions).mean()
+        else:
+            return np.average(self(y_true, raw_predictions),
```

Comment:
I guess we can just call this one? It will be correct when sample_weight=None

Reply:
I kinda did that to avoid any overhead caused by `average`, but it seems the only overhead would be:

    scl = avg.dtype.type(a.size/avg.size)

https://github.com/numpy/numpy/blob/v1.17.0/numpy/lib/function_base.py#L391

So we could just use `average`.